pkg/storage/mvcc.go

```diff
@@ -5935,14 +5955,54 @@ func MVCCFindSplitKey(
 		// Allow a split at any key that sorts after it.
 		minSplitKey = it.UnsafeKey().Key.Clone().Next()
 	}
+	return minSplitKey, nil
+}
 
-	splitKey, err := it.FindSplitKey(key.AsRawKey(), endKey.AsRawKey(), minSplitKey, targetSize)
+// MVCCFirstSplitKey returns the first key which is safe to split at and no
+// less than desiredSplitKey in the range which spans [startKey,endKey). If a
+// non-nil key is returned, it is safe to split at. If a nil key is returned, no
+// safe split key could be determined. The safe split key returned is
+// guaranteed to be:
+//
+//  1. Within [startKey,endKey).
+//  2. No less than desiredSplitKey.
+//  3. Greater than the first key in [startKey,endKey]; or greater than all the
+//     first row's keys if a table range. .
```

Comment:
"range. ."

Feel free to clean this up if you happen to be working in this area on master. No need for the fix to be backported, of course.